desc/utils.py

```diff
@@ -2,14 +2,13 @@
 
 import operator
 import warnings
-from functools import partial
 from itertools import combinations_with_replacement, permutations
 
 import numpy as np
 from scipy.special import factorial
 from termcolor import colored
 
-from desc.backend import flatnonzero, fori_loop, jit, jnp, take
+from desc.backend import flatnonzero, fori_loop, functools, jit, jnp, take
```

Comment:
functools is its own module, why import from backend?

Reply:
no reason, I can change